enhancements/samples-generation.md

```diff
@@ -0,0 +1,613 @@
+---
+title: Automatic Samples Generation using Go language
+authors:
+  - "@camilamacedo86"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-09-02
+last-updated: 2020-09-20
+status: implementable
+---
+
+# Automatic Samples Generation using Go language
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+1. Should not the project has a Memcached GO Sample with webhooks and other without using webhooks since it shows an advanced feature and it used is not mandatory? Note the Go tutorial do not use webhooks.  
```

Comment:
Yes, we would have 2 samples. However, in upstream/kb we have none similar testdata. The idea is we have samples with the steps/implementation suggested to users in the tutorials. The Go Memcached tutorial has none webhooks and we ought to have an advanced tutorial with the additional steps to use webhooks. So, in the end, it would be:

// The Memcached sample without webhooks 
// The Memcached sample with webhooks

Note that we can re-use the first sample steps as we will do in the e2e test to build the second one. 
Also, both should be integrated with OLM since it is a key feature of the tool.